detection_rules/eswrap.py

```diff
@@ -6,37 +6,56 @@
 import json
 import os
 import time
+from collections import defaultdict
+from typing import Union
 
 import click
-from elasticsearch import AuthenticationException, Elasticsearch
+import elasticsearch
+from elasticsearch import Elasticsearch
+from elasticsearch.client import AsyncSearchClient
 
+import kql
 from .main import root
-from .misc import client_error, getdefault
+from .misc import add_params, client_error, elasticsearch_options
 from .utils import format_command_options, normalize_timing_and_sort, unix_time_to_formatted, get_path
+from .rule import Rule
 from .rule_loader import get_rule, rta_mappings
 
 COLLECTION_DIR = get_path('collections')
+MATCH_ALL = {'bool': {'filter': [{'match_all': {}}]}}
 
 
-def get_es_client(es_user, es_password, elasticsearch_url=None, cloud_id=None, **kwargs):
-    """Get an auth-validated elsticsearch client."""
-    assert elasticsearch_url or cloud_id, \
-        'You must specify a host or cloud_id to authenticate to an elasticsearch instance'
+def get_authed_es_client(cloud_id=None, elasticsearch_url=None, es_user=None, es_password=None, ctx=None, **kwargs):
```

Comment:
not a fan of "authed"
```suggestion
def get_elasticsearch_client(cloud_id=None, elasticsearch_url=None, es_user=None, es_password=None, ctx=None, **kwargs):
```

Reply:
wfm - I will make the change for kibana func too